python/mxnet/gluon/data/sampler.py

```diff
@@ -136,3 +136,30 @@ def __len__(self):
         raise ValueError(
             "last_batch must be one of 'keep', 'discard', or 'rollover', " \
             "but got %s"%self._last_batch)
+
+
+class IntervalSampler(Sampler):
+    """Samples elements from [0, length) at fixed intervals.
+
+    Parameters
+    ----------
+    length : int
+        Length of the sequence.
+
+    Examples
+    --------
+    >>> sampler = gluon.data.IntervalSampler(13, interval=3)
+    >>> list(sampler)
+    [0, 3, 6, 9, 12, 1, 4, 7, 10, 2, 5, 8, 11]
```

Comment:
why should it roll over at the end?

Reply:
Is there a reason that you think it shouldn't? I think this sampler should exhaust every sample in a dataset. If for some reason it needs to drop some samples, for the purpose of mini-batch for example, then a wrapper sampler should take care of that.

Reply:
The name interval sampler suggests it should behave like [begin:end:step]

Reply:
I see the confusion now. Should I add an option to specify whether to roll over?

Reply:
This doesn't seem very generic anyway. I would put it in examples

Reply:
This sampler is needed for any long-form text processing that requires passing hidden state from sample to sample. I'd expect repeated use for this, which is why I chose to put it here. Do you prefer to update its interface for handling roll-over, or to move this contrib, or do you still prefer it to be dropped?

Reply:
@piiswrong ping